File: sdk/dotnet-sdk/src/VeritySDK/Protocols/UpdateEndpoint/Common/AuthenticationFactory.cs
using System.Json;
using System.Collections.Generic;
using System.Security.Policy;

namespace VeritySDK.Protocols.UpdateEndpoint
{
    /// <summary>
    /// A factory for a Authentication expression
    /// </summary>
    public class AuthenticationFactory
    {
        /// <summary>
        /// Create authentication parameter for update endpoint which require OAuth2 auth.
        /// This is for version v1 of OAuth2 for verity.
        /// </summary>
        /// <param name="tokenUrl">Url on which token is requested</param>
        /// <param name="data">
        /// Url on which token is requested
        ///     {
        ///         "grant_type": "client_credentials",
        ///         "client_id": "ajeyKDizsDkwYeEmRmHU78gf7W3VIEoA",
        ///         "client_secret": "aaGxxcGi6kb6AxIe"
        ///     }
        /// </param>
        /// <returns>Authentication instance</returns>
        public static Authentication CreateOAuth2V1(Url tokenUrl, Dictionary<string, string> data) {
            JsonObject jsonData = new JsonObject();
            jsonData.Add("url", tokenUrl.ToString());
            foreach(var item in data)
            {
                jsonData.Add(item.Key, item.Value);
            }
            return new Authentication("OAuth2", "v1", jsonData);
        }
    }
}

File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/updateendpoint/common/AuthenticationFactory.java
package com.evernym.verity.sdk.protocols.updateendpoint.common;

import org.json.JSONObject;

import java.net.URL;
import java.util.Map;

public class AuthenticationFactory {
    /**
     * Create authentication parameter for update endpoint which require OAuth2 auth.
     * This is for version v1 of OAuth2 for verity.
     * @param tokenUrl Url on which token is requested
     * @param data additional data required, for eg:
     *         {
     *             "grant_type": "client_credentials",
     *             "client_id": "ajeyKDizsDkwYeEmRmHU78gf7W3VIEoA",
     *             "client_secret": "aaGxxcGi6kb6AxIe"
     *         }
     * @return Authentication instance
     */
    public static Authentication createOAuth2V1(URL tokenUrl, Map<String, String> data) {
        JSONObject jsonData = new JSONObject();
        jsonData.put("url", tokenUrl.toString());
        data.forEach(jsonData::put);
        return new Authentication("OAuth2", "v1", jsonData);
    }
}
